lib/cretonne/src/ir/dfg.rs

```diff
@@ -657,6 +665,28 @@ impl DataFlowGraph {
                         })
     }
 
+    /// Removes the argument `val` from `ebb`. Returns true if the removal is sucessful.
+    pub fn remove_ebb_arg(&mut self, ebb: Ebb, val: Value) -> bool {
```

Comment:
You don't need to update the value table entry for the removed argument, but you need to make sure it is consistent for the remaining arguments.

The value table entry points to the EBB argument list, and the EBB argument list points back to the value table. For attached values, this two-way pointing must be consistent. That is what `is_value_attached` tests. When you remove a value from the EBB argument list, it becomes *detached*, which means that `is_value_attached` returns false.

All of the values that appear in the EBB argument list must be attached, meaning that their value table entries points back to their position in the argument list.